src/main/scala/rise/elevate/rules/traversal.scala

```diff
@@ -58,7 +58,7 @@ object traversal {
     extends Strategy[Rise] {
 
     def apply(e: Rise): RewriteResult[Rise] = e match {
-      case ap @ App(f, x) if f == p => s(x).mapSuccess(App(f, _)(ap.t))
+      case ap @ App(f, x) if f =~= p => s(x).mapSuccess(App(f, _)(ap.t))
```

Comment:
do we traverse untyped expressions sometimes?

Reply:
That seems to be the case, changing it to `=~~=` breaks all of the elevate tests.